mne/time_frequency/tfr.py

```diff
@@ -2207,108 +2429,1110 @@ def plot_joint(
+            return out
+
+        elif dim == "times":
+            return EpochsSpectrum(
+                state,
+                method=None,
+                fmin=None,
+                fmax=None,
+                tmin=None,
+                tmax=None,
+                picks=None,
+                exclude=None,
+                proj=None,
+                remove_dc=None,
+                n_jobs=None,
+            )
+        # ↓↓↓ these two are for dim == "freqs"
+        elif copy:
+            return EpochsTFR(inst=state, method=None, freqs=None)
+        else:
+            self._data = np.expand_dims(data, axis=axis)
+            self._freqs = freqs
+            return self
+
+    @verbose
+    def drop(self, indices, reason="USER", verbose=None):
+        """Drop epochs based on indices or boolean mask.
+
+        .. note:: The indices refer to the current set of undropped epochs
+                  rather than the complete set of dropped and undropped epochs.
+                  They are therefore not necessarily consistent with any
+                  external indices (e.g., behavioral logs). To drop epochs
+                  based on external criteria, do not use the ``preload=True``
+                  flag when constructing an Epochs object, and call this
+                  method before calling the :meth:`mne.Epochs.drop_bad` or
+                  :meth:`mne.Epochs.load_data` methods.
+
+        Parameters
+        ----------
+        indices : array of int or bool
+            Set epochs to remove by specifying indices to remove or a boolean
+            mask to apply (where True values get removed). Events are
+            correspondingly modified.
+        reason : str
+            Reason for dropping the epochs ('ECG', 'timeout', 'blink' etc).
+            Default: 'USER'.
+        %(verbose)s
+
+        Returns
+        -------
+        epochs : instance of Epochs or EpochsTFR
+            The epochs with indices dropped. Operates in-place.
+        """
+        from ..epochs import BaseEpochs
+
+        BaseEpochs.drop(self, indices=indices, reason=reason, verbose=verbose)
+
+        return self
+
+    def iter_evoked(self, copy=False):
+        """Iterate over EpochsTFR to yield a sequence of AverageTFR objects.
+
+        The AverageTFR objects will each contain a single epoch (i.e., no averaging is
+        performed). This method resets the EpochTFR instance's iteration state to the
+        first epoch.
+
+        Parameters
+        ----------
+        copy : bool
+            Whether to yield copies of the data and measurement info, or views/pointers.
+        """
+        self.__iter__()
+        state = self.__getstate__()
+        state["inst_type_str"] = "Evoked"
+        state["dims"] = state["dims"][1:]  # drop "epochs"
+
+        while True:
+            try:
+                data, event_id = self.__next__(return_event_id=True)
+            except StopIteration:
+                break
```

Comment:
This was one at least, I thought at first this could maybe just be:
```
for data, event_id in self:
```
but you wouldn't get the `event_id`. You could also instead do
```
for di, data in enumerate(self):
    event_id = self.events[di, 2]
```
or similar but I'm not sure it's better. It avoids an explicit use of a dunder but also duplicates the `event_id` logic a bit

Reply:
I waffled on this too. I copied this pattern from the similar definition for `Epochs` objects, so I think there's advantage of them being implemented the same way.